src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Yes, exactly. I have updated the PR by clearing the network datasets in `Mle::Stop()`(only for `thread start`) instead of stopping Delay Timer. Thanks.